src/scenic/simulators/webots/model.scenic

```diff
@@ -119,6 +153,89 @@ class Ground(WebotsObject):
             y += dy
         return tuple(heights)
 
+    @staticmethod
+    @distributionFunction
+    def shapeFromHeights(heights, width, length, gridSizeX, gridSizeY, baseThickness):
+        heights = [row[::-1] for row in heights[::-1]]
+
+        triangles = []
+
+        ## Vertices
+        dx, dy = width / (gridSizeX - 1), length / (gridSizeY - 1)
+        base_x, base_y = -width / 2, length / 2
+
+        heightmap_vertices = tuple(tuple((base_x+ix*dx, base_y-iy*dy, heights[iy][ix]) for ix in range(gridSizeX)) for iy in range(gridSizeY))
+
+        raw_vertices = [v for row in heightmap_vertices for v in row]
```

Comment:
Making a big tuple of tuples doesn't seem useful here; why not just use a NumPy array? Then you can use `reshape` to flatten it into a sequence of vertices.